src/app/clusters/time-synchronization-server/time-synchronization-server.cpp

```diff
@@ -0,0 +1,685 @@
+            return aEncoder.Encode(utcTime.count());
+        }
+        else
+        {
+            return aEncoder.EncodeNull();
+        }
+    }
+    case TrustedTimeSource::Id: {
+        return ReadTrustedTimeSource(aPath.mEndpointId, aEncoder);
+    }
+    case DefaultNTP::Id: {
+        return ReadDefaultNtp(aPath.mEndpointId, aEncoder);
+    }
+    case TimeZone::Id: {
+        return ReadTimeZone(aPath.mEndpointId, aEncoder);
+    }
+    case DSTOffset::Id: {
+        return ReadDSTOffset(aPath.mEndpointId, aEncoder);
+    }
+    case TimeZoneListMaxSize::Id: { // can't find a way to initialize default value for F quality
+        uint8_t max = CHIP_CONFIG_TIME_ZONE_LIST_MAX_SIZE;
+        TimeZoneListMaxSize::Set(aPath.mEndpointId, CHIP_CONFIG_TIME_ZONE_LIST_MAX_SIZE);
+        return aEncoder.Encode(max);
+    }
+    case DSTOffsetListMaxSize::Id: {
+        uint8_t max = CHIP_CONFIG_DST_OFFSET_LIST_MAX_SIZE;
+        DSTOffsetListMaxSize::Set(aPath.mEndpointId, CHIP_CONFIG_DST_OFFSET_LIST_MAX_SIZE);
+        return aEncoder.Encode(max);
+    }
+    case LocalTime::Id: {
+        DataModel::Nullable<uint64_t> localTime;
+        if (computeLocalTime(aPath.mEndpointId))
+        {
+            LocalTime::Get(aPath.mEndpointId, localTime);
+            return aEncoder.Encode(localTime.Value());
+        }
+        else
+        {
+            return aEncoder.EncodeNull();
+        }
+    }
+    default: {
+        break;
+    }
+    }
+
+    return err;
+}
+} // anonymous namespace
+
+static bool sendDSTTableEmpty(chip::EndpointId endpointId)
+{
+    Events::DSTTableEmpty::Type event;
+    EventNumber eventNumber;
+
+    CHIP_ERROR error = LogEvent(event, endpointId, eventNumber);
+
+    if (CHIP_NO_ERROR != error)
+    {
+        ChipLogError(Zcl, "Unable to send DSTTableEmpty event [endpointId=%d]", endpointId);
+        return false;
+    }
+    ChipLogProgress(Zcl, "Emit DSTTableEmpty event [endpointId=%d]", endpointId);
+
+    // re-schedule event for after min 1hr
+    // delegate->scheduleDSTTableEmptyEvent()
+    return true;
+}
+
+static bool sendDSTStatus(chip::EndpointId endpointId, bool dstOffsetActive)
+{
+    Events::DSTStatus::Type event;
+    event.DSTOffsetActive = dstOffsetActive;
+    EventNumber eventNumber;
+
+    CHIP_ERROR error = LogEvent(event, endpointId, eventNumber);
+
+    if (CHIP_NO_ERROR != error)
+    {
+        ChipLogError(Zcl, "Unable to send sendDSTStatus event [endpointId=%d]", endpointId);
+        return false;
+    }
+
+    ChipLogProgress(Zcl, "Emit sendDSTStatus event [endpointId=%d]", endpointId);
+    return true;
+}
+
+static bool sendTimeZoneStatus(chip::EndpointId endpointId, uint8_t listIndex)
+{
+    Events::TimeZoneStatus::Type event;
+    auto tz      = TimeSynchronizationServer::Instance().GetTimeZone().begin();
+    event.offset = tz[listIndex].offset;
+    if (tz[listIndex].name.HasValue())
+    {
+        event.name = tz[listIndex].name.Value();
```

Comment:
This doesn't make sense.  Per spec, that field is optional.  Why is it not optional in the XML?